Update trigger doc with new eventConstructor option (#3078)
cypress-io/cypress-documentation: source/api/commands/trigger.md

cy.get('button').trigger('mousemove', { clientX: 200, clientY: 300 })
```

## Fire other Event types.

> By default, `cy.trigger()` fires {% url `Event` https://developer.mozilla.org/en-US/docs/Web/API/Event %}. But for any reason, you might want to trigger other events like `MouseEvent`, `KeyboardEvent`. 
> 
> In that case, use `eventType` option.

Reviewer: ```suggestion
By default, `cy.trigger()` fires {% url `Event` https://developer.mozilla.org/en-US/docs/Web/API/Event %}. But you may want to trigger other events like `MouseEvent` or `KeyboardEvent`. 

In that case, use the `eventType` option.
```

`cy.trigger()` is meant to be a low-level utility that makes triggering events easier than manually constructing and dispatching them. Since any arbitrary event can be triggered, Cypress tries not to make any assumptions about how it should be triggered. This means you'll need to know the implementation details (which may be in a 3rd party library) of the event handler(s) receiving the event and provide the necessary properties.

### Why should I manually set the event type?

> As you can see the documentation of {% url `MouseEvent` https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent %}, most properties of event class instances are read-only. Because of that, it's sometimes impossible to set the value of some properties like `pageX`, `pageY`. It can be problematic in some tests.

Reviewer: ```suggestion
As you can see the documentation of {% url `MouseEvent` https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent %}, most properties of event class instances are read-only. Because of that, it's sometimes impossible to set the value of some properties like `pageX`, `pageY`. This can be problematic in when testing some situations.
```
